src/engine/GroupBy.h

```diff
@@ -276,16 +378,27 @@ class GroupBy : public Operation {
         const ad_utility::AllocatorWithLimit<Id>& alloc,
         const std::vector<HashMapAliasInformation>& aggregateAliases)
         : map_{alloc} {
+      using enum HashMapAggregateType;
       size_t numAggregates = 0;
       for (const auto& alias : aggregateAliases) {
         for (const auto& aggregate : alias.aggregateInfo_) {
           ++numAggregates;
 
-          if (aggregate.aggregateType_ == HashMapAggregateType::AVG)
-            aggregationData_.emplace_back(
-                std::vector<AverageAggregationData>{});
-          if (aggregate.aggregateType_ == HashMapAggregateType::COUNT)
+          if (aggregate.aggregateType_.type_ == AVG)
+            aggregationData_.emplace_back(std::vector<AvgAggregationData>{});
+          if (aggregate.aggregateType_.type_ == COUNT)
             aggregationData_.emplace_back(std::vector<CountAggregationData>{});
+          if (aggregate.aggregateType_.type_ == MIN)
+            aggregationData_.emplace_back(std::vector<MinAggregationData>{});
+          if (aggregate.aggregateType_.type_ == MAX)
+            aggregationData_.emplace_back(std::vector<MaxAggregationData>{});
+          if (aggregate.aggregateType_.type_ == SUM)
+            aggregationData_.emplace_back(std::vector<SumAggregationData>{});
+          if (aggregate.aggregateType_.type_ == GROUP_CONCAT)
+            aggregationData_.emplace_back(
+                std::vector<GroupConcatAggregationData>{});
+
```

Comment:
You can abstract the common code away using a lambda expression, then you just call something like
`addIf.template operator()<CountAggregationData, COUNT>()`
or (with a slightly different implementation)
`addIf(std::type_identity<CountAggregationData>, Count);` (which looks nicer, especially if you give the `std::type_identity` a shorter alias.